src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ShardStateMetricsProcessor.java

```diff
@@ -0,0 +1,108 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.reader;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.AllMetrics;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.PerformanceAnalyzerMetrics;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.reader_writer_shared.Event;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.sql.Connection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NavigableMap;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jooq.BatchBindStep;
+import org.jooq.tools.StringUtils;
+
+public class ShardStateMetricsProcessor implements EventProcessor {
+    private static final Logger LOG = LogManager.getLogger(ShardStateMetricsProcessor.class);
+    private ShardStateMetricsSnapshot shardStateMetricsSnapshot;
+    private BatchBindStep handle;
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+    private static final TypeReference<HashMap<String, String>> TYPE_REF = new TypeReference<HashMap<String, String>>() {};
+
+    private ShardStateMetricsProcessor(ShardStateMetricsSnapshot snapshot) {
+        this.shardStateMetricsSnapshot = snapshot;
+    }
+
+    static ShardStateMetricsProcessor buildShardStateMetricEventsProcessor(
+            long currWindowStartTime,
+            Connection conn,
+            NavigableMap<Long, ShardStateMetricsSnapshot> shardStateEventMetricsMap) {
```

Comment:
nit: Move `ShardStateMetricsSnapshot> shardStateEventMetricsMap)` to next line.

Reply:
Done